airgun/entities/virtwho_configure.py

```diff
@@ -0,0 +1,120 @@
+from navmazing import NavigateToSibling
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep, navigator
+from airgun.views.virtwho_configure import (
+    VirtwhoConfigureCreateView,
+    VirtwhoConfigureEditView,
+    VirtwhoConfiguresView,
+    VirtwhoConfigureDetailsView,
+)
+
+
+class VirtwhoConfigureEntity(BaseEntity):
+
+    def reset_values(self, values):
+        mapping = {
+            'esx': 'VMware vSphere / vCenter (esx)',
+            'xen': 'XenServer (xen)',
+            'hyperv': 'Microsoft Hyper-V (hyperv)',
+            'rhevm': 'Red Hat Virtualization Hypervisor (rhevm)',
+            'libvirt': 'libvirt',
+            'kubevirt': 'Container-native virtualization',
+        }
+        if 'hypervisor_type' in values:
+            hypervisor_value = values['hypervisor_type']
+            values['hypervisor_type'] = mapping[hypervisor_value]
```

Comment:
I'm not sure about this. You are modifying dict in-place, which changes references to dict passed in as `values`.

This might or might not be a problem, depending on how `values` is constructed. But this will be a problem for you, as your `values` is return value of fixture scoped for module (meaning - fixture is run only once and then result is used for every test in entire module).

I think you want to run `values.copy()` and work on copy of `values` dictionary, so your changes don't impact other instances of this dict.

See below for simplified demonstration of problem:
```
$ cat test_problem.py 
from pytest import fixture


@fixture(scope='module')
def form_data():
    form = {
        'debug': True,
        'interval': 'Every hour',
        'hypervisor_id': 'hostname',
        'hypervisor_type': 'esx',
        'hypervisor_content.server': 'localhost'
    }
    return form


def reset_values(values):
    mapping = {
        'esx': 'VMware vSphere / vCenter (esx)',
        'xen': 'XenServer (xen)',
    }
    if 'hypervisor_type' in values:
        hypervisor_value = values['hypervisor_type']
        values['hypervisor_type'] = mapping[hypervisor_value]
    return values


def test_1_first_reset(form_data):
    vals = reset_values(form_data)
    print(vals)


def test_2_dict_is_changed(form_data):
    print(form_data)


def test_3_reset_again(form_data):
    vals = reset_values(form_data)
    print(vals)

$ pytest -sv test_problem.py 
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.3, pytest-4.0.2, py-1.8.0, pluggy-0.12.0 -- /home/mzalewsk/.virtualenvs/robottelo/bin/python3
cachedir: .pytest_cache
rootdir: /tmp, inifile:
plugins: mock-1.10.0, services-1.3.1
collected 3 items                                                                                                                                                                             

test_problem.py::test_1_first_reset {'debug': True, 'interval': 'Every hour', 'hypervisor_id': 'hostname', 'hypervisor_type': 'VMware vSphere / vCenter (esx)', 'hypervisor_content.server': 'localhost'}
PASSED
test_problem.py::test_2_dict_is_changed {'debug': True, 'interval': 'Every hour', 'hypervisor_id': 'hostname', 'hypervisor_type': 'VMware vSphere / vCenter (esx)', 'hypervisor_content.server': 'localhost'}
PASSED
test_problem.py::test_3_reset_again FAILED

========================================================================================== FAILURES ===========================================================================================
_____________________________________________________________________________________ test_3_reset_again ______________________________________________________________________________________

form_data = {'debug': True, 'hypervisor_content.server': 'localhost', 'hypervisor_id': 'hostname', 'hypervisor_type': 'VMware vSphere / vCenter (esx)', ...}

    def test_3_reset_again(form_data):
>       vals = reset_values(form_data)

test_problem.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

values = {'debug': True, 'hypervisor_content.server': 'localhost', 'hypervisor_id': 'hostname', 'hypervisor_type': 'VMware vSphere / vCenter (esx)', ...}

    def reset_values(values):
        mapping = {
            'esx': 'VMware vSphere / vCenter (esx)',
            'xen': 'XenServer (xen)',
        }
        if 'hypervisor_type' in values:
            hypervisor_value = values['hypervisor_type']
>           values['hypervisor_type'] = mapping[hypervisor_value]
E           KeyError: 'VMware vSphere / vCenter (esx)'
```

Reply:
Definitely， this is a problem if there are two cases get form_data and changed by _reset_values. I'd like to update it as:
```
    def _reset_values(self, values):
        mapping = {
            'esx': 'VMware vSphere / vCenter (esx)',
            'xen': 'XenServer (xen)',
            'hyperv': 'Microsoft Hyper-V (hyperv)',
            'rhevm': 'Red Hat Virtualization Hypervisor (rhevm)',
            'libvirt': 'libvirt',
            'kubevirt': 'Container-native virtualization',
        }
        vals = values.copy()
        if 'hypervisor_type' in vals:
            hypervisor_value = vals['hypervisor_type']
            vals.update({'hypervisor_type': mapping[hypervisor_value]})
        return vals
```